src/controllers/admin/accessControl.js

```diff
@@ -0,0 +1,77 @@
+import accessDenied from '../../utils/errorHandling/accessDenied';
+import notFound from '../../utils/errorHandling/notFound';
+import roleServices from '../../services/roles';
+import jwt from 'jsonwebtoken';
+import ApplicationError from '../../utils/applicationError';
+
+
+
+  /* read permissions file */
+  const roles =  roleServices.readFile();
+  let rolesData = {};
+  rolesData = JSON.parse(roles);
+export default  function  permit(permission) {
+
+  return async (req, res, next) => {
+ 
+    try{
+      
+      const userToken = req.header('authorization');
+      if(!userToken){
+        throw new accessDenied('No token found',403);
+      }
+      const tokenVerify = jwt.verify(userToken, process.env.TOKEN_SECRET);
```

Comment:
use the common util fuctions to deal with token

Reply:
Me too, I think so